### ISSUE TITLE
Stop using deprecated Paragon components

### ISSUE THREAD

**Issue by adamstankiewicz** [openedx/frontend-wg#102]
There are 13 deprecated Paragon components; they have been deprecated for awhile now (most, over 2+ years), but we haven't prioritized removing them from Paragon's React component library since they are still in use and don't have great visibility into the impacts on those consumers.

The deprecated components and their replacements, which can be seen on Paragon's [documentation website](https://paragon-openedx.netlify.app/)), are as follows:
* `CheckBoxGroup` (replaced by `Form.CheckboxSet`)
* `Checkbox` (replaced by `Form.Checkbox`)
* `Fieldset` (replaced by `Form.Group`)
* `Input` (replaced by `Form.Control`)
* `InputSelect` (replaced by `Form.Control`)
* `InputText` (replaced by `Form.Control`)
* `ListBox` (turns out, not used anywhere! 🎉)
* `RadioButtonGroup` (replaced by `Form.RadioSet`)
* `TextArea` (replaced by `Form.Control`)
* `ValidationFormGroup` (replaced by `Form.Group`)
* `Modal` (replaced by new modal components, e.g. `AlertModal`, `MarketingModal`, etc.)
  * If there are questions about which modal to use, please reach out to the Paragon Working Group.
* `StatusAlert` (replaced by `Alert`)
* `Table` (replaced by `DataTable`)
  * If there are design related questions, please reach out to the Paragon Working Group.

Once we are confident we can migrate off these deprecated components, the Paragon Working Group will prioritize removing them from the component library in a separate issue.

**Comment by abdullahwaheed** [comment]
[frontend-component-cookie-policy-banner](https://github.com/openedx/frontend-component-cookie-policy-banner) require React and Paragon upgrade for StatusAlert migration

**Comment by abdullahwaheed** [comment]
[studio-frontend](https://github.com/openedx/studio-frontend) requires paragon upgrade

**Comment by hurtstotouchfire** [comment]
@abdullahwaheed fyi the cookie banner will be removed during compliance work soon. It is being replaced by OneTrust.

**Comment by arbrandes** [comment]
@Mashal-m, what's the general situation, here?  Do we have a list of related PRs we can keep track of here?

**Comment by Mashal-m** [comment]
@arbrandes All PRs are linked with this ticket.

**Comment by arbrandes** [comment]
@abdullahwaheed, I only see the Admin Portal PR still open.  Is that the last one?

**Comment by abdullahwaheed** [comment]
> @abdullahwaheed, I only see the Admin Portal PR still open. Is that the last one?

yes, and i thibk owning team is deprecating it themself as Adam commented [here](https://github.com/openedx/frontend-app-admin-portal/pull/872#issuecomment-1938857391)
